api/websocket/server/server.go

```diff
@@ -192,6 +193,24 @@ func (ws *WsServer) registryMethod() {
 			return api.RespPacking(api.NodeInfo(wsAddr, rpcAddr, pubkey, id), errcode.WRONG_NODE)
 		}
 
+		// client auth
+		signature, ok := cmd["Signature"]
+		if ok { // if client send Signature, then check it
+			strSignature := signature.(string)
+			byteSignature, _ := hex.DecodeString(strSignature)
+
+			var challenge []byte
+			sess := cmd["session"].(*session.Session)
+			challenge = sess.Challenge[:]
+
+			err := crypto.Verify(pubKey, challenge, byteSignature)
+			if err != nil { // fail challenge signature
+				return api.RespPacking(nil, errcode.INVALID_PARAMS)
```

Comment:
Better use a new error code. `INVALID_TOKEN` is currently not used, we can rename it and use that